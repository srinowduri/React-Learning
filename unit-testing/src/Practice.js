import React, { Component }  from 'react';

class Practice extends Component{
    state = {
        number: 0
    }
    handleNumber = (event) => {
        this.setState({
            number: event.target.value
        });
    }
    render() {
        return (
            <div>
                <input type="number" onChange={this.handleNumber} />
            </div>
        );
    }
}

export default Practice;