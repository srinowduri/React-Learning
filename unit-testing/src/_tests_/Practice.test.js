import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Practice from '../Practice';

Enzyme.configure({ adapter: new Adapter() });

describe('Practice unit testing', () =>{
    it('should render Practice Component', () => {
        let container = shallow(<Practice />)
        //    console.log(container);
        expect(container.getElements()).toMatchSnapshot();
    })
})