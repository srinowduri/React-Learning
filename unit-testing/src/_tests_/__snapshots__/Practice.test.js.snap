// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Practice unit testing should render Practice Component 1`] = `
Array [
  <div>
    <input
      onChange={[Function]}
      type="number"
    />
  </div>,
]
`;
